dotnet/src/Connectors/Connectors.GoogleVertexAI/Extensions/VertexAIKernelBuilderExtensions.cs

```diff
@@ -0,0 +1,256 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Net.Http;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.SemanticKernel.ChatCompletion;
+using Microsoft.SemanticKernel.Connectors.GoogleVertexAI;
+using Microsoft.SemanticKernel.Embeddings;
+using Microsoft.SemanticKernel.Http;
+using Microsoft.SemanticKernel.TextGeneration;
+
+namespace Microsoft.SemanticKernel;
+
+/// <summary>
+/// Extensions for adding VertexAI generation services to the application.
+/// </summary>
+public static class VertexAIKernelBuilderExtensions
+{
+    /// <summary>
+    /// Adds Vertex AI Gemini Text Generation service to the kernel builder.
+    /// </summary>
+    /// <param name="builder">The kernel builder.</param>
+    /// <param name="modelId">The model for text generation.</param>
+    /// <param name="bearerKeyProvider">The Bearer Key provider for authentication.</param>
+    /// <param name="location">The location to process the request</param>
+    /// <param name="projectId">Your project ID</param>
+    /// <param name="serviceId">The optional service ID.</param>
+    /// <param name="httpClient">The optional custom HttpClient.</param>
+    /// <returns>The updated kernel builder.</returns>
+    /// <remarks>
+    /// This <paramref name="bearerKeyProvider"/> will be called on every request,
+    /// when providing the token consider using caching strategy and refresh token logic
+    /// when it is expired or close to expiration.
+    /// </remarks>
+    public static IKernelBuilder AddVertexAIGeminiTextGeneration(
+        this IKernelBuilder builder,
+        string modelId,
+        Func<string> bearerKeyProvider,
```

Comment:
@Krzysztof318, I agree with the first point of Sergey on the BearerKey namning, we may consider renaming the parameter to `BearerToken` instead of key what you think @SergeyMenshykh?

Reply:
@RogerBarreto I have already done that in PR which you merged :)